helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -210,6 +213,18 @@ private void updateStateTransitionMessageThreadPool(Message message, HelixManage
       return;
     }
 
+    if (!_batchMessageThreadpoolChecked) {
```

Comment:
It seems to me as long as there's any `STATE_TRANSITION` message, this pool will be created, so why not simply instantiate this thread pool in the constructor?